src/knx/knx_value.cpp

```diff
@@ -1,562 +1,562 @@
+        case TimeType:
+            return (int64_t)ulongValue();
+        case CharType:
+            return (int64_t)_value.charValue;
+        case ShortType:
+            return (int64_t)_value.shortValue;
+        case IntType:
+            return (int64_t)_value.intValue;
+        case DoubleType:
+            return (int64_t)_value.doubleValue;
+        case StringType:
+#ifndef KNX_NO_STRTOx_CONVERSION
+            return strtol(_value.stringValue, NULL, 0);
+#else
+            return 0;
+#endif
+    }
+    return 0;
+}
+
+double KNXValue::doubleValue() const
+{
+    switch (_type)
+    {
+        case DoubleType:
+            return _value.doubleValue;
+         case BoolType:
+            return _value.boolValue ? 1 : 0;
+        case UCharType:
+            return _value.ucharValue;
+        case UShortType:
+            return _value.ushortValue;
+        case UIntType:
+            return _value.uintValue;
+        case ULongType:
+            return _value.uintValue;
+        case TimeType:
+            return ulongValue();
+        case CharType:
+            return _value.charValue;
+        case ShortType:
+            return _value.shortValue;
+        case IntType:
+            return _value.intValue;
+        case LongType:
+            return _value.longValue;
+        case StringType:
+#ifndef KNX_NO_STRTOx_CONVERSION
+            return strtod(_value.stringValue, NULL);
+#else
+            return 0;
+#endif
+    }
+    return 0;
+}
+
+const char* KNXValue::stringValue() const
+{
+    switch (_type)
+    {
+        case DoubleType:
+        case BoolType:
+        case UCharType:
+        case UShortType:
+        case UIntType:
+        case ULongType:
+        case TimeType:
+        case CharType:
+        case ShortType:
+        case IntType:
+        case LongType:
+            return ""; // we would have to manage the memory for the string otherwise. Maybe later.
+        case StringType:
+            return _value.stringValue;
+    }
+    return 0;
+}
+
+struct tm KNXValue::timeValue() const
+{
+    switch (_type)
+    {
+        case TimeType:
+            return _value.timeValue;
+        case BoolType:
+        case UCharType:
+        case UShortType:
+        case UIntType:
+        case ULongType:
+        case CharType:
+        case ShortType:
+        case IntType:
+        case LongType:
+        case DoubleType:
+        case StringType:
+        {
+            time_t timeVal = ulongValue();
+            struct tm timeStruct;
+            gmtime_r(&timeVal, &timeStruct);
+            return timeStruct;
+        }
+    }
+    struct tm tmp = {0};
+    return tmp;
+}
+
+KNXValue::KNXValue(float value)
+{
+    _value.doubleValue = value;
+    _type = DoubleType;
+}
+
+KNXValue& KNXValue::operator=(const float value)
+{
+    _value.doubleValue = value;
+    _type = DoubleType;
+    return *this;
+}
+
+KNXValue::operator float() const
+{
+    return doubleValue();
 }
```

Comment:
Did you change anything here? Where the line endings wrong?

Reply:
No, nothing changed as far as I know. Was surprised to see it being included in the push.